public/i18n.js

```diff
@@ -11,10 +11,45 @@ const locales = {
   "es-ES": "i18n/es-es.json"
 };
 
-$(function () {
+const getConfigForLocale = (locale) => {
+  if (!locale) {
+    return null;
+  }
+
+  if (locales[locale]) {
+    return {
+      locale: locale,
+      map: {
+        [locale]: locales[locale]
+      }
+    };
+  }
+
+  const language = locale.split('-')[0];
+
+  if (language && locales[language]) {
+    return {
+      locale: language,
+      map: {
+        [language]: locales[language]
+      }
+    };
+  }
+
+  return null;
+};
+
+const determineLocaleConfig = (detectedLocale) => {
   const searchParams = new URLSearchParams((new URL(window.location)).search);
-  const locale = searchParams.get('locale');
+  const localeParam = searchParams.get('locale');
+
+  // first try locale selected in app by user
+  // next try locale detected by jQuery.i18n library
+  // if all else fails, give them English
+  return getConfigForLocale(localeParam) || getConfigForLocale(detectedLocale) || { en: locales.en };
```

Comment:
Fallback {en: locales.en} has a different signature than what getConfigForLocales returns. Might be cleaner to pass an array of locales to geConfigForLocale and have it return the locale+map for the first entry that exists.